src/clj/rems/db/applications.clj

```diff
@@ -451,6 +451,23 @@
            "")
    :maxlength (:maxlength item)})
 
+(defn- keyword-keys-to-numbers [[k v]]
+  [(if (keyword? k)
+     (Integer/parseInt (name k))
+     k)
+   v])
+
+(defn- assoc-item-previous-values [application items]
+  (let [previous-values (->> (if (editable? (:state application))
+                               (:submitted-form-contents application)
+                               (:previous-submitted-form-contents application))
+                             :items
+                             ;; XXX: due to event deserialization, the keys are no more numbers after reading from database
```

Comment:
this fixing could be done when fetching the events from the db? see fix-event-from-db